src/model_selection/crossValScore.ts

```diff
@@ -0,0 +1,210 @@
+ */
+export async function crossValScore<
+  T extends {
+    fit(X: Tensor2D, y: Tensor1D): Promise<unknown>
+  }
+>(
+  estimator: T,
+  Xy: [Tensor2D, Tensor1D],
+  params: {
+    cv?: CrossValidator
+    groups?: Tensor1D
+    scoring: (this: T, X: Tensor2D, y: Tensor1D) => Scalar
+  }
+): Promise<Tensor1D>
+
+/**
+ * Evaluates a score by cross-validation. This particular overload
+ * of the function uses the given scorer to cross validate an
+ * unsuperwised estimator.
+ *
+ * @param estimator A supervised estimator that has an async `fit(X)` function.
+ *
+ * @param X        An single element array containing the features of the
+ *                 cross validation dataset.
+ *
+ * @param params   Cross validation parameters. `cv` is the {@link CrossValidator}
+ *                 responsible for splitting the dataset into training in test
+ *                 data in one or more folds. `groups` is additional grouping
+ *                 information which is used by some {@link CrossValidator}
+ *                 instance to determine which groups of datapoints may not
+ *                 appear both in test and training data at the same time.
+ *                 `scoring` is the scorer function which is being called
+ *                 with `estimator` as `this` and the test features `X_test`
+ *                 as argument.
+ *
+ * @returns        The test score for each split/fold that was generated by
+ *                 the {@link CrossValidator} `cv`
+ */
+export async function crossValScore<
+  T extends {
+    fit(X: Tensor2D): Promise<unknown>
+  }
+>(
+  estimator: T,
+  X: [Tensor2D],
+  params: {
+    cv?: CrossValidator
+    groups?: Tensor1D
+    scoring: (this: T, X: Tensor2D) => Scalar
+  }
+): Promise<Tensor1D>
+
+/**
+ * Evaluates a score by cross-validation. This particular overload
+ * of the function uses the default score of a superwised estimator
+ * for scoring.
+ *
+ * @param estimator A supervised estimator that has an async `fit(X,y)`
+ *                 function and a `score(X,y)` function.
+ *
+ * @param Xy       A two element array containing the features and targets
+ *                 of the cross validation dataset.
+ *
+ * @param params   Cross validation parameters. `cv` is the {@link CrossValidator}
+ *                 responsible for splitting the dataset into training in test
+ *                 data in one or more folds. `groups` is additional grouping
+ *                 information which is used by some {@link CrossValidator}
+ *                 instance to determine which groups of datapoints may not
+ *                 appear both in test and training data at the same time.
+ *
+ * @returns        The test score for each split/fold that was generated by
+ *                 the {@link CrossValidator} `cv`
+ */
+export async function crossValScore(
+  estimator: {
+    fit(X: Tensor2D, y: Tensor1D): Promise<unknown>
+    score(X: Tensor2D, y: Tensor1D): Scalar
+  },
+  Xy: [Tensor2D, Tensor1D],
```

Comment:
Better to separate these params?

Reply:
The difficulty is that this method is supposed to work for unsupervised and supervised models. With supervised models the call would be:
```js
crossValScore(supervisedEstimator, X, y, {...})
```
And in the supervised case it would be:
```js
crossValScore(unsupervisedEstimator, X, {...})
```
But since the parameters are intended to be optional, the following call would also be valid:
```js
crossValScore(supervisedEstimator, X, y)
```
How to then decide whether y is a `Scikit1D` or a set of parameters? That is why I packed `[X, y]` into an array... but maybe there is a better way?

Reply:
Oh, I see. I always go with an intuitive API since JavaScript does not support named parameters. That is all required params are positional and all optional params are part of the Param object. 

So in this case something like:

```javascript
crossValScore(unsupervisedEstimator | supervisedEstimator, X, { y, ...})
```

Another suggestion would be to make the second param an object, so the order does not matter. Example:
```javascript
 crossValScore(unsupervisedEstimator | supervisedEstimator, { X, y }, {...})
```

@DirkToewe @dcrescim  Thoughts?

Reply:
It should work as intended now. Both `crossValsScore(unsupervised, X, undefined, {})` and `crossValsScore(unsupervised, X, {})` should work.